File: 0001-two-sum/0001-two-sum.cpp
#include <vector>

class Solution {
public:
    std::vector<int> twoSum(std::vector<int>& nums, int target) {
        int n = nums.size();
        std::vector<int> ans;

        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                if (nums[i] + nums[j] == target) {
                    ans.push_back(i);
                    ans.push_back(j);
                 // Stop the loop and return the result
                }
            }
        }

        // If no solution is found
        return ans;
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dct={}
        
        for i in range(len(nums)):
            if target-nums[i] in dct:
                return [dct[target-nums[i]],i]
            dct[nums[i]]=i
# class Solution:
#     def twoSum(self, arr: List[int], target: int) -> List[int]:
#         n=len(arr)
#         res={}
#         for i,num in enumerate(arr):
#             complement=target-num
#             if complement in res:
#                return [res[complement],i]
#             res[num]=i
    